ui/app/components/mount-backend-form.ts

```diff
@@ -49,6 +50,14 @@ export default class MountBackendForm extends Component<Args> {
 
   @tracked errorMessage: string | string[] = '';
 
+  constructor(owner: unknown, args: Args) {
+    super(owner, args);
+
+    if (!this.args.mountType) {
```

Comment:
Love the assertion! 

The [typical pattern I've seen us do is using](https://github.com/hashicorp/vault/blob/main/ui/lib/core/addon/components/form-field.js#L83) `assert` - I'm not sure if it makes much of a difference, but it is slightly cleaner by not needing to wrap it in a conditional.

```js
assert( `@mountType is required. Must be "auth" or "secret".`', presence(this.args.mountType);